File: sicpa-root/sicpa-dto/src/main/java/com/sicpa/demo/dto/DepartmentDto.java
package com.sicpa.demo.dto;

import java.io.Serializable;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class DepartmentDto extends BaseDto implements Serializable {

	private static final long serialVersionUID = 1L;
	private Long id;
	@NotEmpty
	private String description;
	@NotEmpty
	private String name;
	private String phone;
	@NotNull
	private EnterpriseDto enterprise;
	//private List<DepartmentEmployeeDto> departmentEmployees;
	
}

File: sicpa-web/src/app/models/department-dto.ts
import { DepartmentEmployeeDto } from "./department-employee-dto";
import { EnterpriseDto } from "./enterprise-dto";
import { GenericDto } from "./generic-dto";

export class DepartmentDto extends GenericDto {

  id: number;
  description: string;
  name: string;
  phone: string;
  enterprise: EnterpriseDto= new EnterpriseDto();
  //departmentEmployees: Array<DepartmentEmployeeDto>;

  constructor() {
    super();
  }
}
